src/tests/query.rs

```diff
@@ -1955,6 +1955,20 @@ async fn coalesce_fun(api: &mut dyn TestApi) -> crate::Result<()> {
     Ok(())
 }
 
+fn value_into_json(value: &Value) -> Option<serde_json::Value> {
+    match value.clone() {
+        // MariaDB returns JSON as text
+        Value::Text(Some(text)) => {
+            let json: serde_json::Value =
+                serde_json::from_str(&text).expect(format!("expected parsable text to json, found {}", text).as_str());
+
+            Some(json)
+        }
+        Value::Json(Some(json)) => Some(json),
+        _ => None,
+    }
+}
```

Comment:
I ended up using an external helper and not adding a match to `Value::as_json()` because it's a const method and `serde_json::from_str` is not so it cannot work. And changing the const fn to a simple fn I believe is a breaking change that wasn't worth it.